src/torrent/connection_manager.h

```diff
@@ -22,41 +22,45 @@ class LIBTORRENT_EXPORT ConnectionManager {
   using port_type     = uint16_t;
   using priority_type = uint8_t;
 
-  static constexpr priority_type iptos_default     = 0;
-  static constexpr priority_type iptos_lowdelay    = IPTOS_LOWDELAY;
-  static constexpr priority_type iptos_throughput  = IPTOS_THROUGHPUT;
-  static constexpr priority_type iptos_reliability = IPTOS_RELIABILITY;
+  enum iptos : priority_type {
+    def         = 0,
+    lowdelay    = IPTOS_LOWDELAY,
+    throughput  = IPTOS_THROUGHPUT,
+    reliability = IPTOS_RELIABILITY,
 
 #if defined IPTOS_MINCOST
-  static constexpr priority_type iptos_mincost     = IPTOS_MINCOST;
+    mincost     = IPTOS_MINCOST,
 #elif defined IPTOS_LOWCOST
-  static constexpr priority_type iptos_mincost     = IPTOS_LOWCOST;
+    mincost     = IPTOS_LOWCOST,
 #else
-  static constexpr priority_type iptos_mincost     = iptos_throughput;
+    mincost     = iptos::throughput,
 #endif
+  };
 
-  static constexpr uint32_t encryption_none             = 0;
-  static constexpr uint32_t encryption_allow_incoming   = (1 << 0);
-  static constexpr uint32_t encryption_try_outgoing     = (1 << 1);
-  static constexpr uint32_t encryption_require          = (1 << 2);
-  static constexpr uint32_t encryption_require_RC4      = (1 << 3);
-  static constexpr uint32_t encryption_enable_retry     = (1 << 4);
-  static constexpr uint32_t encryption_prefer_plaintext = (1 << 5);
+  enum encryption : uint32_t {
+    none             = 0,
+    allow_incoming   = (1 << 0),
+    try_outgoing     = (1 << 1),
+    require          = (1 << 2),
+    require_RC4      = (1 << 3),
+    enable_retry     = (1 << 4),
+    prefer_plaintext = (1 << 5),
 
   // Internal to libtorrent.
-  static constexpr uint32_t encryption_use_proxy        = (1 << 6);
-  static constexpr uint32_t encryption_retrying         = (1 << 7);
-
-  enum {
-    handshake_incoming           = 1,
-    handshake_outgoing           = 2,
-    handshake_outgoing_encrypted = 3,
-    handshake_outgoing_proxy     = 4,
-    handshake_success            = 5,
-    handshake_dropped            = 6,
-    handshake_failed             = 7,
-    handshake_retry_plaintext    = 8,
-    handshake_retry_encrypted    = 9
+    use_proxy        = (1 << 6),
+    retrying         = (1 << 7),
+  };
+
+  enum handshake {
```

Comment:
These are unscoped enums, so `ConnectionManager::incoming` is also valid.

You need to use `enum class {`